src/player/Player.php

```diff
@@ -1216,6 +1216,13 @@ public function getXpDropAmount() : int{
 		return 0;
 	}
 
+	public function onEnchant(int $enchantLevel) : void{
+		if($this->hasFiniteResources()){
+			$this->xpManager->subtractXpLevels($enchantLevel);
+		}
+		$this->xpSeed = $this->generateXpSeed();
+	}
```

Comment:
This function seems redundant. The logic can be done directly in `EnchantTransaction`.

In addition, a public `regenerateXpSeed() : void` would probably be a good idea, to allow plugins to re-roll the XP seed.

Reply:
Maybe allow plugins to setXpSeed?

Reply:
Yeah, that might not be a bad idea actually. Might be useful for reproducing bugs.